src/libData/BlockData/Block/MicroBlock.cpp

```diff
@@ -81,6 +81,15 @@ MicroBlock::MicroBlock(const MicroBlockHeader& header,
   m_cosigs = move(cosigs);
 }
 
+MicroBlock::MicroBlock(const MicroBlockHeader& header,
+                       const vector<TxnHash>& tranHashes)
+    : m_header(header), m_tranHashes(tranHashes) {
+  if (m_header.GetNumTxs() != m_tranHashes.size()) {
```

Comment:
We can probably have a more useful message here, like displaying `GetNumTxs` and `size` values.